src/BTree.mo

```diff
@@ -0,0 +1,110 @@
+/// Imperative sequences as B-Trees.
+
+import A "Array";
+import I "Iter";
+import List "List";
+import Option "Option";
+import Order "Order";
+import P "Prelude";
+import Prim "mo:⛔";
+
+module {
+
+  /// Constants we use to shape the tree.
+  /// See https://en.wikipedia.org/wiki/B-tree#Definition
+  module Constants {
+    let MAX_CHILDREN = 4;
+  };
+
+  public type Compare<K> = {
+    compare : (K, K) -> Order.Order
+  };
+
+  public type Data<K, V> = [(K, V)];
```

Comment:
Curious, what's the benefit of the `Data<K, V>` type being the tuple `[(K, V)]` over something like

```
public type Data<K, V> = {
  key: K;
  value: V;
};
```

Reply:
There's a slight space savings to the tuple (1 word per instance).  Otherwise, I'd prefer the record with labeled fields.

I actually had that same record definition initially, but recalled [this recent review from Claudio for the HashMap improvements](https://github.com/dfinity/motoko-base/pull/394#pullrequestreview-1015044080).

Reply:
[The Rust implementation in `std`](https://github.com/rust-lang/rust/blob/8c1cc82a82ebfbea731258063115992b2dd4adee/library/alloc/src/collections/btree/node.rs#L9)